internal/daemonset/daemonset.go

```diff
@@ -254,7 +270,11 @@ func (dc *daemonSetGenerator) SetDevicePluginAsDesired(ctx context.Context, ds *
 
 	serviceAccountName := mod.Spec.DevicePlugin.ServiceAccountName
 	if serviceAccountName == "" {
-		serviceAccountName = rbac.GenerateDevicePluginServiceAccountName(*mod)
+		if useDefaultSA {
+			serviceAccountName = "kmm-operator-device-plugin"
+		} else {
+			log.FromContext(ctx).Info("Warning: no ServiceAccount set for the device plugin DaemonSet")
```

Comment:
Same note as above: maybe we want to fail here

Reply:
I don't think so.

Reply:
ok, then let's use util.WarnString API for warning messages